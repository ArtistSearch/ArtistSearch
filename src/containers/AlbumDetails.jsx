import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { getAlbumInfo } from '../services/getArtists';
import AlbumSongList from '../components/songs/AlbumSongList';
import Loading from '../components/Loading';
import style from '../components/songs/AlbumSongList.css'

export default function AlbumDetails() {
  const [loading, setLoading] = useState(true);
  const [songList, setSongList] = useState([]);

  const { artistName, releaseId, title } = useParams();

  useEffect(() => {
    getAlbumInfo(releaseId)
      .then(setSongList)
      .finally(() => setLoading(false));
  }, []);

  if (loading) return (
    <div className={style.trackListWrapper}>
      <Loading />
    </div>);

  return (
    <section>
      <AlbumSongList
        songs={songList}
        art={releaseId}
        artist={artistName}
        title={title}
      />
    </section>
  );
}
