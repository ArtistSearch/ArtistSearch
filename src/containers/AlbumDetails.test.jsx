import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import AlbumDetails from './AlbumDetails';
import { MemoryRouter, Route } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import releaseResData from '../data/tracklistForTest.json';

const server = setupServer(
  rest.get(
    'https://musicbrainz.org/ws/2/recording?release=0ec052ee-8c50-48f3-8682-501778dfc823&fmt=json',
    (req, res, ctx) => {
      return res(ctx.json(releaseResData));
    }
  )
);

describe('Detail page Container Component', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays a list of the songs on the album', async () => {
    render(
      <MemoryRouter
        initialEntries={[
          '/release/cc197bad-dc9c-440d-a5b5-d52ba2e14234/0ec052ee-8c50-48f3-8682-501778dfc823',
        ]}
      >
        <Route path="/release/:artistName/:releaseId/">
          <AlbumDetails />
        </Route>
      </MemoryRouter>
    );

    const trackList = await screen.findByRole('list', {
      name: 'Album Track List',
    });
    expect(trackList).toMatchSnapshot();
    expect(trackList).not.toBeEmptyDOMElement();

    // const loadingScreen = await screen.getByText('Loading...');
    // expect(loadingScreen).toMatchSnapshot();

    // const titleHeading = await screen.findByRole('heading', {
    //   name: 'Song Title',
    // });
    // expect(titleHeading).toMatchSnapshot();

    return waitFor(() => {
      screen.getByText(/Sparks/);
      screen.getByAltText('album cover art');
    });
  });
});
