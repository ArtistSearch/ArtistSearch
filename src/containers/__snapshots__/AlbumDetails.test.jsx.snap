// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail page Container Component displays a list of the songs on the album 1`] = `
<ul
  aria-label="Album Track List"
  class="albumInfo"
>
  <a
    href="/song/cc197bad-dc9c-440d-a5b5-d52ba2e14234/Trouble"
  >
    <li>
      <h4
        aria-label="Song Title"
      >
        Trouble
      </h4>
    </li>
  </a>
  <a
    href="/song/cc197bad-dc9c-440d-a5b5-d52ba2e14234/Sparks"
  >
    <li>
      <h4
        aria-label="Song Title"
      >
        Sparks
      </h4>
    </li>
  </a>
  <a
    href="/song/cc197bad-dc9c-440d-a5b5-d52ba2e14234/Parachutes"
  >
    <li>
      <h4
        aria-label="Song Title"
      >
        Parachutes
      </h4>
    </li>
  </a>
  <a
    href="/song/cc197bad-dc9c-440d-a5b5-d52ba2e14234/Spies"
  >
    <li>
      <h4
        aria-label="Song Title"
      >
        Spies
      </h4>
    </li>
  </a>
  <a
    href="/song/cc197bad-dc9c-440d-a5b5-d52ba2e14234/Shiver"
  >
    <li>
      <h4
        aria-label="Song Title"
      >
        Shiver
      </h4>
    </li>
  </a>
  <a
    href="/song/cc197bad-dc9c-440d-a5b5-d52ba2e14234/Yellow"
  >
    <li>
      <h4
        aria-label="Song Title"
      >
        Yellow
      </h4>
    </li>
  </a>
  <a
    href="/song/cc197bad-dc9c-440d-a5b5-d52ba2e14234/Everything’s Not Lost / Life Is for Living"
  >
    <li>
      <h4
        aria-label="Song Title"
      >
        Everything’s Not Lost / Life Is for Living
      </h4>
    </li>
  </a>
  <a
    href="/song/cc197bad-dc9c-440d-a5b5-d52ba2e14234/We Never Change"
  >
    <li>
      <h4
        aria-label="Song Title"
      >
        We Never Change
      </h4>
    </li>
  </a>
  <a
    href="/song/cc197bad-dc9c-440d-a5b5-d52ba2e14234/High Speed"
  >
    <li>
      <h4
        aria-label="Song Title"
      >
        High Speed
      </h4>
    </li>
  </a>
  <a
    href="/song/cc197bad-dc9c-440d-a5b5-d52ba2e14234/Don’t Panic"
  >
    <li>
      <h4
        aria-label="Song Title"
      >
        Don’t Panic
      </h4>
    </li>
  </a>
</ul>
`;
